packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -570,14 +570,14 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
       this.props.onKeyDown(ev);
     }
 
-    const { onMenuClick } = this.props;
-    if (onMenuClick) {
-      onMenuClick(ev, this);
-    }
-
     if (!ev.defaultPrevented &&
-      this.props.menuTriggerKeyCode !== null &&
+      this.props.menuTriggerKeyCode &&
```

Comment:
It doesn't seem like you want to check `this.props.menuTriggerKeyCode` here? It is already being handled in `this._isValidMenuOpenKey`.